examples/autopoint/MultidimensionalGaussian_mixture_autopoint.cpp

```diff
@@ -0,0 +1,126 @@
+// Use forward-mode automatic differentiation using Autodiff Library
+
+// VolEsti (volume computation and sampling library)
+
+// Copyright (c) 2012-2020 Vissarion Fisikopoulos
+// Copyright (c) 2018-2020 Apostolos Chalkis
+// Copyright (c) 2020-2020 Marios Papachristou
+// Copyright (c) 2022-2022 Zhang zhuyan
+
+// Contributed and/or modified by Marios Papachristou, as part of Google Summer of Code 2020 program.
+// Contributed and/or modified by Zhang zhuyan, as part of Google Summer of Code 2020 program.
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+
+#include <iostream>
+#include <cmath>
+#include <functional>
+#include <vector>
+#include <unistd.h>
+#include <string>
+#include <typeinfo>
+
+#include "Eigen/Eigen"
+
+#include "ode_solvers/ode_solvers.hpp"
+#include "random.hpp"
+#include "random/uniform_int.hpp"
+#include "random/normal_distribution.hpp"
+#include "random/uniform_real_distribution.hpp"
+#include "random_walks/random_walks.hpp"
+#include "volume/volume_sequence_of_balls.hpp"
+#include "volume/volume_cooling_gaussians.hpp"
+#include "volume/volume_cooling_balls.hpp"
+#include "generators/known_polytope_generators.h"
+#include "readData.h"
+#include "diagnostics/diagnostics.hpp"
+
+
+template <typename NT>
+void run_main()
+{
+    typedef Cartesian<NT> Kernel;
+    typedef typename Kernel::Point Point;
+    typedef std::vector<Point> pts;
+    typedef HPolytope<Point> Hpolytope;
+    typedef BoostRandomNumberGenerator<boost::mt19937, NT> RandomNumberGenerator;
+    typedef AutoDiffFunctor::GradientFunctor<Point> NegativeGradientFunctor;
+    typedef AutoDiffFunctor::FunctionFunctor<Point> NegativeLogprobFunctor;
+    typedef LeapfrogODESolver<Point, NT, Hpolytope, NegativeGradientFunctor> Solver;
+    typedef typename Hpolytope::MT    MT;
+    typedef typename Hpolytope::VT    VT;
+    AutoDiffFunctor::parameters<NT> params;
+    params.data=readMatrix<NT>("data.txt");
+    NegativeGradientFunctor F(params);
+    NegativeLogprobFunctor f(params);
+    RandomNumberGenerator rng(1);
+    unsigned int dim = 4;
+
+    HamiltonianMonteCarloWalk::parameters<NT, NegativeGradientFunctor> hmc_params(F, dim);
+    std::chrono::time_point<std::chrono::high_resolution_clock> start,stop;
+
+    Hpolytope P = generate_cube<Hpolytope>(dim, false);
+
+    Point x0 = -0.25 * Point::all_ones(dim);
+
+    // In the first argument put in the address of an H-Polytope
+    // for truncated sampling and NULL for untruncated
+    HamiltonianMonteCarloWalk::Walk<Point, Hpolytope, RandomNumberGenerator, NegativeGradientFunctor, NegativeLogprobFunctor, Solver>
+        hmc(&P, x0, F, f, hmc_params);
+    int n_samples = 50000; // Half will be burned
+    int max_actual_draws = n_samples / 2;
+    unsigned int min_ess = 0;
+    MT samples;
+    samples.resize(dim, max_actual_draws);
+
```

Comment:
If you add a line hmc.eta = some number (eg. 0.0001) that should manually set the learning rate

Reply:
added